### PR TITLE
Setting redeem date from door time based on hours before showtime

### LINKED ISSUE #459: Setting the time a ticket can be scanned for an event
The event `redeem_date` needs to be set somewhere on the event page

### PR DESCRIPTION
**Pull request by sdbondi**
Added redeemDate - please check the logic for restricting the door time and redeem date fields

Closes #459

### CODE REVIEW

**Comment by zmelamed** [comment]
Yes I think it makes sense that the door time should be used to set redeem time. 

Regarding the null, I believe we may be better served having the org member select the number of hours  prior to the show as the door time rather than a time stamp. The default should be 1 hour before showtime. We should move the show time before the doortime on the form so that gets set first

I don’t think we should have open ended checkins times though .

**Comment by sdbondi** [comment]
@zmelamed Cool - do you see that as a drop down saying something like 

- At door time
- 1 hour
- 2 hours
etc

or a number input for number of hours before door time?

File: src/components/common/form/SelectGroup.js [hunk 1]
```diff
@@ -26,24 +29,23 @@ const SelectGroup = props => {
...
+			<MenuItem key={item.value} value={item.value}>
+				{item.label || item.value}
 			</MenuItem>
 		));
 	}
 
 	return (
 		<FormControl
-			className={classes.formControl}
+			className={classes[styleClassName]}
```
Comment (sdbondi) [inline src/components/common/form/SelectGroup.js]: @Jasonvdb Style hack to fix margin pushing select down

**Comment by sdbondi** [comment]
@zmelamed @Jasonvdb @Krakaw PR is ready - 

Please see if how this works is 💯 

screen shot 2018-12-21 at 13 43 56

screen shot 2018-12-21 at 13 42 49

**Comment by zmelamed** [comment]
There doesn’t need to be two separate inputs for door time and redeem time. It should be event Event Date, Show Time, then next to that would be the Door Time which would be the dropdown you created and that would define the door time which also be the redeem time.

**Comment by sdbondi** [comment]
@zmelamed Pushed these updates - let me know if you're happy with the dropdown phraseology etc 

screen shot 2018-12-24 at 07 43 04

screen shot 2018-12-24 at 07 43 11

**Pull request merged**